COMPATIBILITY.md

```diff
@@ -0,0 +1,18 @@
+- [Compatibility with OpenSearch](#compatibility-with-opensearch)
+
+## Compatibility with OpenSearch
+
+The below matrix shows the compatibility of the `opensearch-net` with versions of [`OpenSearch`](https://opensearch.org/downloads.html#opensearch). Currently this client is in a pre-release phase. The below matrix shows compatibility for the upcoming version.
+
+| OpenSearch Version | Client Version |
+| --- | --- |
+| 1.0.0 | 1.0.0 |
+| 1.0.1 | 1.0.0 |
+| 1.1.0 | 1.0.0 |
+| 1.2.0 | 1.0.0 |
+| 1.2.1 | 1.0.0 |
+| 1.2.2 | 1.0.0 |
+| 1.2.3 | 1.0.0 |
+| 1.2.4 | 1.0.0 |
+| 1.3.0 | 1.0.0 |
+| 1.3.1 | 1.0.0 |
```

Comment:
would it be better to use `1.x.x` or `1.2.x` instead of listing all versions? since there shouldn't be breaking changes from OpenSearch in the same major version

Reply:
Sure, for now I kept it consistent with the other clients since some clients have 1.1 version as well. We could update it once the matrix becomes large enough.